internal/dag/httpproxy_processor.go

```diff
@@ -533,6 +533,25 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 		return
 	}
 	insecure.CORSPolicy = cp
+	dynamicHeaders := map[string]string{
+		"CONTOUR_NAMESPACE": proxy.Namespace,
+	}
+
+	requestHeadersPolicy, err := headersPolicyRoute(proxy.Spec.VirtualHost.RequestHeadersPolicy, false, dynamicHeaders)
+	if err != nil {
```

Comment:
lets add some negative test cases for if validation fails on virtualhost specified header policies

Reply:
this can go alongside the other test added in `builder_test.go`